lib/commands/samples.ts

```diff
@@ -4,26 +4,36 @@
 import path = require("path");
 var options: any = require("../common/options");
 
-export class PrintSamplesInformationCommand implements ICommand {
-	constructor(private $samplesService: ISamplesService) { }
+class PrintSamplesCommand implements ICommand {
+	constructor(private $samplesService: ISamplesService,
+		private framework?: string) { }
+
 	execute(args: string[]): IFuture<void> {
-		return (() => {
-			this.$samplesService.printSamplesInformation().wait();
-		}).future<void>()();
+		return this.$samplesService.printSamplesInformation(this.framework);
 	}
+
 	allowedParameters: ICommandParameter[] = []
 }
-$injector.registerCommand("sample|*list", PrintSamplesInformationCommand);
+var frameworkIdentifiers: Project.ITargetFrameworkIdentifiers = $injector.resolve("projectConstants").TARGET_FRAMEWORK_IDENTIFIERS;
+var subCommands = {
+	"*list": "",
+	"hybrid": frameworkIdentifiers.Cordova,
+	"native": frameworkIdentifiers.NativeScript,
+	"website": frameworkIdentifiers.MobileWebsite
+};
+var samplesService = $injector.resolve("samplesService");
+_.each(subCommands, (subCommand, framework) => {
+	$injector.registerCommand("sample|" + subCommand, new PrintSamplesCommand(samplesService, framework));
+});
```

Comment:
I suggest you to use something similar to this service https://github.com/Icenium/icenium-cli/blob/master/lib/services/project-commands-service.ts#L8. I also think it should be great if you create `CommandsServiceBase` and reuse the logic.